client/main.lua

```diff
@@ -55,49 +55,50 @@ local function checkType(ped, animType)
     local model = GetEntityModel(ped)
     local animConfig = config[animType]
     local drawableType = GetPedDrawableVariation(ped, animConfig.drawable)
-    local textureType = GetPedTextureVariation(ped, drawableType)
+    local textureType = GetPedTextureVariation(ped, animConfig.drawable)
 
     if lib.table.contains(config.blacklistedPeds, model) then
         TriggerEvent("chat:addMessage", "[Radio Anims] The ped model you're using is blacklisted!")
         return false
     end
 
     for gender, modelType in pairs(modelTypes) do
-        for _, drawableVariation in pairs(animConfig[gender].variations) do
-            table.insert(drawableVariationType, drawableVariation)
-        end
-
-        for _, textureVariation in pairs(animConfig[gender].textures) do
-            table.insert(textureVariationType, textureVariation)
+        if model == GetHashKey(modelType) then
+            drawableVariationType = animConfig[gender].variations
+            textureVariationType = animConfig[gender].textures
```

Comment:
Function: Directly assigns the corresponding variations and textures to drawableVariationType and textureVariationType if the ped model matches the expected model type.

Advantages:
-Accuracy: Only the relevant variations and textures are assigned based on the ped model.
-Efficiency: There is no need to use table.insert as the variations and textures are assigned directly.
-Clarity: The code is clearer and easier to understand because it directly checks if the model matches before applying the variations and textures.